src/editors/editorsController.ts

```diff
@@ -0,0 +1,45 @@
+import * as vscode from 'vscode';
+
+import ConnectionController from '../connectionController';
+import CollectionDocumentsProvider, {
+  NAMESPACE_URI_IDENTIFIER,
+  VIEW_COLLECTION_SCHEME
+} from './collectionDocumentsProvider';
+import { createLogger } from '../logging';
+
+const log = createLogger('editors controller');
+
+/**
+ * This controller manages when our extension needs to open
+ * new editors and the data they need. It also manages active editors.
+ */
+export default class EditorsController {
+  activate(connectionController: ConnectionController): void {
+    log.info('activating...');
+    const collectionViewProvider = new CollectionDocumentsProvider(connectionController);
+
+    vscode.workspace.registerTextDocumentContentProvider(
+      VIEW_COLLECTION_SCHEME, collectionViewProvider
+    );
+
+    log.info('activated.');
+  }
+
+  onViewCollectionDocuments(namespace): Thenable<void> {
+    const uriQuery = `?${NAMESPACE_URI_IDENTIFIER}=${namespace}`;
+    // We attach the current time to ensure a new editor window is opened on
```

Comment:
Have you thought about `cursor.id` vs. timestamp?

Reply:
Hmmm... I think for now since we're just showing 10 documents and closing the query using `timestamp` makes the most sense. 

Once we're opening and storing a cursor for pagination using code lenses (hopefully) then using an identifier for the cursor might make sense. I'm not sure we have an `id` on a cursor we can use though? https://mongodb.github.io/node-mongodb-native/3.5/api/Cursor.html#~toArrayResultCallback Maybe I'm not following.

We also could make the timestamp string a readable string, but I found it ended up as a longer than we want to show.